lndsigner_test.go

```diff
@@ -380,13 +379,60 @@ func TestIntegration(t *testing.T) {
 		resp = lnds[1].Lncli(t, ctx, "payinvoice", "-f", invoice)
 	})
 
+	t.Run("get invoice from lnd2 for lnd1", func(t *testing.T) {
+		resp = lnds[1].Lncli(t, ctx, "addinvoice", "5000")
+		invoice = resp["payment_request"].(string)
+	})
+
+	t.Run("pay invoice from lnd1 to lnd2", func(t *testing.T) {
+		resp = lnds[0].Lncli(t, ctx, "payinvoice", "-f", invoice)
+	})
+
+	t.Run("get invoice from lnd3 for lnd2", func(t *testing.T) {
+		resp = lnds[2].Lncli(t, ctx, "addinvoice", "5000")
+		invoice = resp["payment_request"].(string)
+	})
+
+	t.Run("pay invoice from lnd2 to lnd3", func(t *testing.T) {
+		resp = lnds[1].Lncli(t, ctx, "payinvoice", "-f", invoice)
+	})
+
+	t.Run("get invoice from lnd2 for lnd3", func(t *testing.T) {
+		resp = lnds[1].Lncli(t, ctx, "addinvoice", "5000")
+		invoice = resp["payment_request"].(string)
+	})
```

Comment:
You define these as sub-tests, but it feels that that is a bit too granular perhaps?

Reply:
I've refactored this and it's less granular. Hopefully the new arrangement makes better sense.

Reply:
Yes looks good